corehq/motech/repeaters/migrations/0016_add_indexes.py

```diff
@@ -0,0 +1,100 @@
+from django.db import migrations, models
+
+
+class Migration(migrations.Migration):
+    atomic = False
+
+    dependencies = [
+        ("repeaters", "0015_repeater_max_workers"),
+    ]
+
+    operations = [
+        migrations.SeparateDatabaseAndState(
+            state_operations=[
+                migrations.AlterField(
+                    model_name="repeater",
+                    name="is_deleted",
+                    field=models.BooleanField(default=False),
+                ),
+                migrations.AddIndex(
+                    model_name="repeater",
+                    index=models.Index(
+                        condition=models.Q(("is_deleted", False)),
+                        fields=["id"],
+                        name="deleted_partial_idx",
+                    ),
+                ),
+                migrations.AddIndex(
+                    model_name="repeater",
+                    index=models.Index(
+                        condition=models.Q(
+                            ("is_deleted", False),
+                            ("is_paused", False),
+                        ),
+                        fields=["id"],
+                        name="deleted_paused_partial_idx",
+                    ),
+                ),
+                migrations.AddIndex(
+                    model_name="repeatrecord",
+                    index=models.Index(
+                        condition=models.Q(("state__in", (1, 2))),
+                        fields=["repeater_id"],
+                        name="state_partial_idx",
+                    ),
+                ),
+            ],
+
+            database_operations=[
+                # Drop `Repeater.id_deleted` index
+                migrations.RunSQL(
+                    sql="""
+                    DROP INDEX CONCURRENTLY "repeaters_repeater_is_deleted_08441bf0";
+                    """,
+                    reverse_sql="""
+                    CREATE INDEX CONCURRENTLY "repeaters_repeater_is_deleted_08441bf0"
+                        ON "repeaters_repeater" ("is_deleted");
+                    """
+                ),
+
+                # Replace with a partial index on `id_` column. Used
+                # when next_attempt_at is null
```

Comment:
I think "Used when next_attempt_at is null" may or may not be true, depending on the structure of the query and what data is in the table. Seems like it might be true for some environments but not others.